documentation/modules/adding-hook-using-cli.adoc

```diff
@@ -0,0 +1,130 @@
+// Module included in the following assemblies:
+//
+// * documentation/doc-Migration_Toolkit_for_Virtualization/master.adoc
+
+:_content-type: PROCEDURE
+[id="adding-migration-hook-via-cli_{context}"]
+= Adding a migration hook to a migration plan using the CLI
+
+You can add a migration hook to an existing migration plan using the {project-first} CLI.
+
+You can add a `PreHook` or a `PostHook` Hook CR when you migrate a virtual machine from the command line by using the {project-full} API. A `PreHook` runs before a migration, a `PostHook`, after.
+
+[NOTE]
+====
+You can retrieve additional information stored in a secret or in a `configMap` by using a `k8s` module.
+====
+
+For example, you can create a hook CR to install `cloud-init` on a VM and write a file before migration.
+
+.Prerequisites
+
+* Migration plan
+* Migration hook image or the playbook containing the hook image
+
+[NOTE]
+====
+You can run one pre-migration hook, one post-migration hook, or one of each, per migration plan.
+====
+
+* File containing the `Secret` for the source provider
+* {ocp} service account that has at least write-access permission for the namespace you are working in
+* SSH access for VMs you want to migrate with the public key installed on the VMs
+* VMs running on Microsoft Server only: Remote Execution enabled
+
+.Procedure
+. If needed, create a Secret with an SSH private key for the VM. You can either use an existing key or generate a key pair, install the public key on the VM, and base64 encode the private key in the Secret.
```

Comment:
Maybe: If needed, create a Secret with an SSH private key for the VM.

1. (sub-steps) Choose an existing key or generate a key pair.
2. Install the public key on the VM.
3. Encode the private key in the secret to base64.